lib/linter-plus.coffee

```diff
@@ -31,12 +31,17 @@ class Linter
       @commands.lint()
 
     @subscriptions.add atom.workspace.observeTextEditors (editor) =>
-      currentEditorLinter = new EditorLinter @, editor
-      @editorLinters.set editor, currentEditorLinter
-      @emitter.emit 'observe-editor-linters', currentEditorLinter
-      currentEditorLinter.lint false
-      editor.onDidDestroy =>
-        currentEditorLinter.destroy()
+      editorLinter = new EditorLinter(editor)
+      @editorLinters.set editor, editorLinter
+      @emitter.emit 'observe-editor-linters', editorLinter
+      editorLinter.onShouldUpdateBubble =>
+        @views.renderBubble()
+      editorLinter.onShouldUpdateLineMessages =>
+        @views.updateLineMessages(true)
+      editorLinter.onShouldLint =>
```

Comment:
Because the linters are stored on the linter object and instead of changing or accessing your neighbours, I found this one to be easy to debug and specific. I mean, editor stuff and subscriptions are stored in editor-linter and linter stuff like addLinter, deleteLinter, lint and locking and stuff is managed by `linter-registry`, You can check it out in https://github.com/AtomLinter/Linter/pull/737